pkg/api/vlabs/validate.go

```diff
@@ -1667,6 +1667,10 @@ func (k *KubernetesConfig) validateNetworkPlugin(hasWindows bool) error {
 		return errors.Errorf("networkPlugin '%s' is not supporting windows agents", networkPlugin)
 	}
 
+	if networkPlugin == NetworkPluginKubenet && hasWindows {
```

Comment:
Let's unit test this using the super cool `Example`-style UT. See `ExampleProperties_validateLocation` as an example in `validate_test.go`.